src/Microsoft.AspNetCore.OData/Query/EnableQueryAttribute.cs

```diff
@@ -542,7 +542,7 @@ private ODataQueryOptions CreateAndValidateQueryOptions(HttpRequest request, ODa
         {
             RequestQueryData requestQueryData = request.HttpContext.Items[nameof(RequestQueryData)] as RequestQueryData;
 
-            if (requestQueryData.QueryValidationRunBeforeActionExecution)
+            if (requestQueryData?.QueryValidationRunBeforeActionExecution ?? false)
```

Comment:
It took me a bit of time to understand this change. An easier to fix would be:
```suggestion
            if (requestQueryData != null && requestQueryData.QueryValidationRunBeforeActionExecution)
```